static/js/upload.js

```diff
@@ -92,19 +97,47 @@ export function get_item(key, config) {
 
 export function hide_upload_status(config) {
     get_item("send_button", config).prop("disabled", false);
-    get_item("send_status", config).removeClass("alert-info").hide();
+    get_item("send_status", config).remove();
+}
+
+function show_upload_banner(config, banner_type, banner_text) {
+    // We only show one upload banner at a time per compose box,
+    // and all uploads are combined into the same progress bar.
+    // TODO: It would be nice to separate the error banner into
+    // a different element, so that we can show it at the same
+    // time as the upload bar and other uploads can still continue
+    // when an error occurs.
+    const $upload_banner = get_item("send_status", config);
+    if ($upload_banner.length) {
+        if (banner_type === "error") {
+            // Hide moving bar so that it doesn't do the 1s transition to 0
+            const $moving_bar = $(`${get_item("send_status_identifier", config)} .moving_bar`);
+            $moving_bar.hide();
+            $upload_banner.removeClass("info").addClass("error");
+            // Show it again once the animation is complete.
+            setTimeout(() => $moving_bar.show(), 1000);
```

Comment:
Can you explain what this line is about? I worry about it doing weird things like re-showing the bar after the message is sent or the region has been dismissed, but maybe that's fine because an outer element will have been hidden?

Reply:
Hmm, it was to make sure it was visible for the future if the container becomes an upload (non-error) banner again, but I could also do one of these:

* render a new banner when transitioning from error to upload
* make sure the moving bar is visible when transitioning from error to upload

Reply:
Hmm. Ultimately, we're going to want to render a new moving_bar for each file being uploaded, rather than being a singleton, right? So possibly that tweak would provide an opportunity to delete this.

Reply:
Yup that will let us delete it! The error bar being separate also simplifies it.